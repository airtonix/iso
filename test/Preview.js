const React = require('react')
const { create: render } = require('react-test-renderer')
const Preview = require('../src/Preview').default

const fm = `---
title: hello
---
<h1>beep</h1>
`

describe('Preview', () => {
  test('renders', () => {
    const json = render(
      React.createElement(Preview, {
        filename: 'hello.jsx',
        code: '<h1>hello</h1>',
        scope: {},
        store: {}
      })
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('handles invalid jsx', () => {
    const json = render(
      React.createElement(Preview, {
        filename: 'hello.jsx',
        code: 'h1>invalid</h1>',
        scope: {},
        store: {}
      })
    ).toJSON()
    expect(json).toBe(null)
  })

  test('renders with SCProvider', () => {
    const json = render(
      React.createElement(Preview, {
        filename: 'hello.jsx',
        code: '<h1>hello</h1>',
        scope: {},
        theme: {},
        store: {
          styledComponents: true
        }
      })
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with webfont link', () => {
    const json = render(
      React.createElement(Preview, {
        filename: 'hello.jsx',
        code: '<h1>hello</h1>',
        scope: {},
        theme: {
          font: '"Roboto", sans-serif'
        },
        store: {}
      })
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with front matter', () => {
    const json = render(
      React.createElement(Preview, {
        filename: 'hello.jsx',
        code: fm,
        scope: {},
        theme: {},
        store: {}
      })
    ).toJSON()
    expect(json).toMatchSnapshot()
  })
})
