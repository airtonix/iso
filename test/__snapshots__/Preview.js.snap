// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview renders 1`] = `
<div>
  <h1>
    hello
  </h1>
</div>
`;

exports[`Preview renders with SCProvider 1`] = `
<div
  className="sc-bdVaJa cUmmwg"
  color="#000"
>
  <h1>
    hello
  </h1>
</div>
`;

exports[`Preview renders with front matter 1`] = `
<div>
  <h1>
    beep
  </h1>
</div>
`;

exports[`Preview renders with webfont link 1`] = `
Array [
  <link
    href="https://fonts.googleapis.com/css?family=Roboto"
    rel="stylesheet"
  />,
  <div>
    <h1>
      hello
    </h1>
  </div>,
]
`;
